pkg/bless/oidc.go

```diff
@@ -0,0 +1,192 @@
+	payload, err := json.Marshal(signingRequest)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not json marshal payload")
+	}
+	return o.getCert(ctx, payload)
+}
+
+func (o *OIDC) getCert(ctx context.Context, payload []byte) (*ssh.Certificate, error) {
+	responseBytes, err := o.awsClient.Lambda.ExecuteWithQualifier(
+		ctx,
+		o.lambdaConfig.FunctionName,
+		o.lambdaConfig.FunctionVersion,
+		payload,
+	)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Raw lambda response %s", string(responseBytes))
+	lambdaResponse := &LambdaResponse{}
+	err = json.Unmarshal(responseBytes, lambdaResponse)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not deserialize lambda reponse")
+	}
+
+	logrus.Debugf("Parsed lambda response %s", spew.Sdump(lambdaResponse))
+	if lambdaResponse.ErrorType != nil {
+		if lambdaResponse.ErrorMessage != nil {
+			return nil, errors.Errorf("bless error: %s: %s", *lambdaResponse.ErrorType, *lambdaResponse.ErrorMessage)
+		}
+		return nil, errors.Errorf("bless error: %s", *lambdaResponse.ErrorType)
+	}
+
+	if lambdaResponse.Certificate == nil {
+		return nil, errors.New("No certificate in response")
+	}
+
+	certBytes, err := base64.StdEncoding.DecodeString(*lambdaResponse.Certificate)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not b64 decode certificate")
+	}
+
+	pubKey, err := ssh.ParsePublicKey(certBytes)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not parse public key from lambda response")
+	}
+
+	cert, ok := pubKey.(*ssh.Certificate)
+	if !ok {
+		return nil, errors.Errorf("lambda return type not *ssh.Certificate")
+	}
+
+	return cert, nil
+}
+
+// SigningRequest is a request for a certificate
+// TODO(el): copy/paste from ssh-ca-lambda. Use that once open source
+type SigningRequest struct {
+	RemoteUsernames RemoteUsernames  `json:"remote_usernames,omitempty"`
+	PublicKeyToSign *PublicKeyToSign `json:"public_key_to_sign,omitempty"`
+
+	// IdentityAssertion used to verify the caller
+	Identity Identity `json:"identity,omitempty"`
+}
+
+// Identity represents different types of identity assertions
+// that we can use
+type Identity struct {
+	OktaAccessToken *OktaAccessTokenInput `json:"okta_identity,omitempty"`
+}
+
+type OktaAccessTokenInput struct {
+	AccessToken string
+}
+
+type RemoteUsernames []string
+
+// String returns the string representation of RemoteUsernames
+func (ru RemoteUsernames) String() string {
+	return strings.Join(ru, ",")
+}
+
+// List returns the []string representation of RemoteUsernames
+func (ru RemoteUsernames) List() []string {
+	return ru
+}
+
+func (ru RemoteUsernames) MarshalJSON() ([]byte, error) {
+	return json.Marshal(ru.String())
+}
+
+func (ru RemoteUsernames) UnmarshalJSON(data []byte) error {
+	var remoteUsernames string
+	err := json.Unmarshal(data, &remoteUsernames)
+	if err != nil {
+		return errors.Wrap(err, "error unmarshalling remote usernames")
+	}
+	ru = RemoteUsernames(strings.Split(remoteUsernames, ","))
```

Comment:
**[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
ineffectual assignment to `ru` (ineffassign)